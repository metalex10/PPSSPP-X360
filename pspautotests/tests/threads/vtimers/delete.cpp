#include "shared.h"

inline void testDelete(const char *title, SceUID vtimer) {
	int result = sceKernelDeleteVTimer(vtimer);
	if (result == 0) {
		checkpoint("%s: OK", title);
	} else {
		checkpoint("%s: Failed (%08x)", title, result);
	}
}

SceUInt deleteHandler(SceUID uid, SceInt64 scheduled, SceInt64 actual, void *common) {
	checkpoint("Delete within handler: %08x", sceKernelDeleteVTimer((SceUID)common));
	return 0;
}

extern "C" int main(int argc, char *argv[]) {
	SceUID vtimer = sceKernelCreateVTimer("delete", NULL);

	testDelete("Normal", vtimer);
	testDelete("NULL", 0);
	testDelete("Invalid", 0xDEADBEEF);
	testDelete("Deleted", vtimer);

	vtimer = sceKernelCreateVTimer("delete", NULL);
	sceKernelStartVTimer(vtimer);
	testDelete("Started", vtimer);

	vtimer = sceKernelCreateVTimer("delete", NULL);
	sceKernelSetVTimerHandlerWide(vtimer, 1, &deleteHandler, (void *)vtimer);
	sceKernelStartVTimer(vtimer);
	sceKernelDelayThread(10000);
	checkpoint("Delete after handler: %08x", sceKernelDeleteVTimer(vtimer));

	return 0;
}