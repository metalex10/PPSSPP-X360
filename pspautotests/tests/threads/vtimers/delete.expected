[x] Normal: OK
[x] NULL: Failed (800201be)
[x] Invalid: Failed (800201be)
[x] Deleted: Failed (800201be)
[x] Started: OK
[r] Delete within handler: 80020064
[x] Delete after handler: 00000000
